### DISCUSSION TITLE
Welcome to HoodLoader2 Discussions!

### DISCUSSION THREAD

**Discussion by NicoHood** [NicoHood/HoodLoader2#83] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by raoulbea-fr** [comment]
bonjour, as tu deja essayer de remettre le firmware original avec avrdude? 
si oui peux tu me dire comment faire je ne comprend pas le fonctionnement.
j'ai une elegoo méga2560 r3 et hoodloader2 me supprime les ports analogique alors que j'en ai besoin pour mon projet.
merci de m'avoir lu.

**Reply by NicoHood** [reply]
Please write your questions in english

**Reply by raoulbea-fr** [reply]
hello, have you ever tried to reset the original firmware with avrdude?
if yes can you tell me how to do i don't understand how it works.
i have an elegoo mega2560 r3 and hoodloader2 removes my analog ports while i need them for my project.
Thanks for reading me.

**Reply by NicoHood** [reply]
You just install them as you would install hoodloader2.

But "hoodloader2 removes my analog ports" makes no sense to me. There is nothing that should influence the 2560 behavior.

**Comment by raoulbea-fr** [comment]
hi, i think i don't master such a powerful firmware i want to go back to the factory firmware but i can't do it i'm sorry to bother you do you have a procedure?
when I try to put the factory firmware I have the message
avrdude: stk500v2_ReceiveMessage (): timeout
thanks for your time.

**Reply by mikewax** [reply]
there are many references about how to do that. 
https://docs.arduino.cc/hacking/software/DFUProgramming8U2
https://www.youtube.com/watch?v=fSXZMVdO5Sg

**Comment by customsolutions** [comment]
I like your work on Hoodloader and really enjoy getting into the nuts and bolts of things.  As a project to do so I would like to load a hoodloader esc firmware(16u2 version), on a 32u4 use as the 16u2 but with more abilities....  I have been playing with getting HL2 on both a 16u2/328p(uno) combo as well as a leonardo32u4 configuration.... Is the 32u4 hex listed ported from a 16u2 so to operate the same. or is it more of a modified version of caterina, or am I really comparing apples to apples with out realizing it....  This is at the cusp of my understanding and Im trying to determine the operational differences between a usb to serial/oem 16u2 fw/ caterina/ HL2 of the appropriate versions. I would love to do my own compile and or mods, but atmel studio/ Lufa and the above are at the edge of my current abilities. I would like to learn.... Thanks in advance for any info/heads up...

**Reply by NicoHood** [reply]
I dont really know what to answer. The 32u4 firmware is fully tested and functional. It works similar to the 16u2 firmware.

In general all those firmwares work mostly the same, the difference is that I've made it smaller, so it fits an additional usb to serial function.

**Reply by customsolutions** [reply]
Thanks for your reply, im assuming that the usb to serial also interprets or listens for commands from the ide at upload time, so as to communicate with bootloader on 328, or so as to load on local chip(16u2/32u4). And obviously the reset pulse from user to select which chip?

**Reply by NicoHood** [reply]
I really dont understand you question.

There is a special upload baud rate, at which the bootloader does not forward the data to the 328 but rather reprograms itself.

**Reply by customsolutions** [reply]
Just want to update/share.... i had designed my own leonardo(micro)board. Which is breaded and connected successfully with a 328p and is running hoodloader2.

**Comment by razvitm** [comment]
Hi, I sent you an e-mail about this also, but just in case you don't receive it...
I designed a PCB layout the size of an arduino nano that uses a 16U2 ... What do you think?
![image](https://user-images.githubusercontent.com/93677343/140174416-5b37358d-16b1-4a76-b671-64de88e8f3cc.png)

**Reply by NicoHood** [reply]
I answered your email yesterday:

Hey Razvan!
Looks like a very nice Design!

The question is, what is your usecase? Why not use a Pro Micro Board and flash the Hoodloader2 there as well? It has a more powerful 32u4, and Hoodloader2 also supports this MCU.

Cheers
Nico

**Reply by razvitm** [reply]
At work we buy mega1280 boards and desolder the 1280 and use it on our boards (because of silicon shortage) and I am left with 30-40 boards which still have the 16U2 on them, and I didn't want to throw these away so I designed the Arduino Nano 16U2 board that is going to use hoodloader and the salvaged 16U2s from the dfrobot arduino mega 1280 boards. PS I did not receive your email. PPS it was in the spam folder.

**Reply by NicoHood** [reply]
Makes sense then. Why not. Sounds like a nice idea, especially when you have all pins broken out. Actually then you do not even need hoodloader2, a "normal" cdc bootloader like the lufa bootloader would also work. But that requires setting up the whole thing first and you also need to apply my fix, that is still not merged.

So yeah, give it a try :-)
